### DISCUSSION TITLE
Fan control

### DISCUSSION THREAD

**Discussion by ZhVladyslav** [BartoszCichecki/LenovoLegionToolkit#924] [Ideas]
The idea is to add the ability to adjust the rotation of the coolers within a certain temperature for each of the laptop's operating modes (quiet, balance, performance and custom).

Of the utilities that I managed to find, this functionality is implemented only in the LegionFanControl utility.

The reason for this need is that the default settings of the cooler revolutions are quite bad and are the same for all modes with the exception of the revolutions that are reached at the maximum temperature in each of the modes. Also, the standard minimum revolutions in all modes are 1800 at a temperature of 45, although at 1000 revolutions and light loads, the temperature is maintained stably within these limits, but the characteristic hum from the coolers disappears, which is quite noticeable for me personally at 1800 revolutions (if the revolutions are higher, the hum disappears), and the maximum the revolutions in the modes (quiet, balance) do not reach the maximum if the temperature is very high.

**Comment by BartoszCichecki** [comment]
This has been discussed many times previously. Nothing more than currently implemented is possible with Lenovo provided API.
